concepts/reduce/about.md

```diff
@@ -0,0 +1,75 @@
+# About
```

Comment:
Given that this is a more complicated concept, I wonder if it makes sense to start with an example before giving the syntax and technical bits. Maybe start with the motivation? Something like this?

> Suppose we want to sum an array of numbers. In many languages, this is expressed as a loop with an accumulator variable. This can also be written as a recursive function, `def sum(x): (x | first) + sum(x | rest)`. This is a common pattern which can be described as a reducing function.

Reply:
> Given that this is a more complicated concept, I wonder if it makes sense to start with an example before giving the syntax and technical bits. Maybe start with the motivation? Something like this?
> 
> > Suppose we want to sum an array of numbers. In many languages, this is expressed as a loop with an accumulator variable. This can also be written as a recursive function, `def sum(x): (x | first) + sum(x | rest)`. This is a common pattern which can be described as a reducing function.

How about this:

---
> Suppose we want to sum an array of numbers. There are many ways to accomplish this goal.
> 
> - In many languages, this problem is expressed as a loop with an accumulator variable.
> - This problem can be written as a recursive function. In pseudo-code, this might look like
> 
>   ```none
>   function add(x, sum=0):
>     if x is empty
>       return sum
>     else
>       return add(rest(x), sum + first(x))
>   ```
> 
>   This method of dividing the problem into smaller pieces can also be described as "reducing towards the base case."
> 
> **Reduce** is a way to ...

Reply:
This looks solid. Though I'm not a fan of sentences without any terminations prior to code blocks :slightly_smiling_face:

Reply:
That's a bad habit. Will fix in the next commit.